packages/postcss-svgo/src/index.js

```diff
@@ -53,71 +52,50 @@ function minifyPromise(decl, getSvgo, opts) {
       return;
     }
 
-    promises.push(
-      getSvgo()
-        .optimize(svg)
-        .then((result) => {
-          let data, optimizedValue;
-
-          if (isBase64) {
-            data = Buffer.from(result.data).toString('base64');
-            optimizedValue = 'data:image/svg+xml;base64,' + data + url.hash;
-          } else {
-            data = isUriEncoded ? encode(result.data) : result.data;
-            // Should always encode # otherwise we yield a broken SVG
-            // in Firefox (works in Chrome however). See this issue:
-            // https://github.com/cssnano/cssnano/issues/245
-            data = data.replace(/#/g, '%23');
-            optimizedValue = 'data:image/svg+xml;charset=utf-8,' + data;
-            quote = isUriEncoded ? '"' : "'";
-          }
-
-          node.nodes[0] = Object.assign({}, node.nodes[0], {
-            value: optimizedValue,
-            quote: quote,
-            type: 'string',
-            before: '',
-            after: '',
-          });
-        })
-        .catch((error) => {
-          throw new Error(`${PLUGIN}: ${error}`);
-        })
-    );
+    const result = optimize(svg, opts);
+    if (result.error) {
+      throw new Error(`${PLUGIN}: ${result.error}`);
```

Comment:
There are possible exception from optimize call which probably need to be wrapped too. This is something I missed why developing v2. v3 will probably only throw.

Reply:
Do you mean it's best to call `optimize()` inside a `try/catch` ?

Reply:
I mean both handle result.error and with try/catch or like this
```
let reuslt
try {
  result = optmize(svg, opts)
  if (result.error) {
    throw Error(result.error)
  }
} catch (error) {
  throw Error(`${PLUGIN}: ${error.message}`)
}
```